smartsim/_core/generation/operations.py

```diff
@@ -0,0 +1,298 @@
+import base64
+import os
+import pathlib
+import pickle
+import sys
+import typing as t
+from dataclasses import dataclass, field
+
+from ..commands import Command
+
+entry_point_path = "smartsim._core.entrypoints.file_operations"
+"""Path to file operations module"""
+
+copy_cmd = "copy"
+"""Copy file operations command"""
+symlink_cmd = "symlink"
+"""Symlink file operations command"""
+configure_cmd = "configure"
+"""Configure file operations command"""
+
+
+def create_final_dest(
+    job_root_path: pathlib.Path, dest: t.Union[pathlib.Path, None]
+) -> str:
```

Comment:
Similar "can we mark this protected" question here
```suggestion
def _create_final_dest(
    job_root_path: pathlib.Path, dest: t.Union[pathlib.Path, None]
) -> str:
```